certmanager/generate_cert.go

```diff
@@ -83,6 +83,32 @@ const (
 // See http://www.alvestrand.no/objectid/2.5.29.17.html.
 var oidSubjectAltName = asn1.ObjectIdentifier{2, 5, 29, 17}
 
+// GenCSR generates a X.509 certificate sign request with the given options.
+func GenCsr(options CertOptions) ([]byte, []byte) {
+	// Generates a CSR
+	// The public key will be bound to the certficate generated below. The
+	// private key will be used to sign this certificate in the self-signed
+	// case, otherwise the certificate is signed by the signer private key
+	// as specified in the CertOptions.
```

Comment:
Some of the comments are not appropriate here - this method should only generate a CSR and not worry about who is going to sign it.